bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxLocalAccessHandler.java

```diff
@@ -0,0 +1,172 @@
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_CURRENT,
+                new QuantityType<>(data.getInverterCurrent(), Units.AMPERE));
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_VOLTAGE,
+                new QuantityType<>(data.getInverterVoltage(), Units.VOLT));
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_FREQUENCY,
+                new QuantityType<>(data.getInverterFrequency(), Units.HERTZ));
+
+        updateState(SolaxBindingConstants.INVERTER_PV1_POWER, new QuantityType<>(data.getPV1Power(), Units.WATT));
+        updateState(SolaxBindingConstants.INVERTER_PV1_CURRENT, new QuantityType<>(data.getPV1Current(), Units.AMPERE));
+        updateState(SolaxBindingConstants.INVERTER_PV1_VOLTAGE, new QuantityType<>(data.getPV1Voltage(), Units.VOLT));
+
+        updateState(SolaxBindingConstants.INVERTER_PV2_POWER, new QuantityType<>(data.getPV2Power(), Units.WATT));
+        updateState(SolaxBindingConstants.INVERTER_PV2_CURRENT, new QuantityType<>(data.getPV2Current(), Units.AMPERE));
+        updateState(SolaxBindingConstants.INVERTER_PV2_VOLTAGE, new QuantityType<>(data.getPV2Voltage(), Units.VOLT));
+
+        updateState(SolaxBindingConstants.INVERTER_PV_TOTAL_POWER,
+                new QuantityType<>(data.getPVTotalPower(), Units.WATT));
+        updateState(SolaxBindingConstants.INVERTER_PV_TOTAL_CURRENT,
+                new QuantityType<>(data.getPVTotalCurrent(), Units.AMPERE));
+
+        updateState(SolaxBindingConstants.BATTERY_POWER, new QuantityType<>(data.getBatteryPower(), Units.WATT));
+        updateState(SolaxBindingConstants.BATTERY_CURRENT, new QuantityType<>(data.getBatteryCurrent(), Units.AMPERE));
+        updateState(SolaxBindingConstants.BATTERY_VOLTAGE, new QuantityType<>(data.getBatteryVoltage(), Units.VOLT));
+        updateState(SolaxBindingConstants.BATTERY_TEMPERATURE,
+                new QuantityType<>(data.getBatteryTemperature(), SIUnits.CELSIUS));
+        updateState(SolaxBindingConstants.BATTERY_STATE_OF_CHARGE,
+                new QuantityType<>(data.getBatterySoC(), Units.PERCENT));
+
+        updateState(SolaxBindingConstants.FEED_IN_POWER, new QuantityType<>(data.getFeedInPower(), Units.WATT));
+
+        updateState(SolaxBindingConstants.TIMESTAMP, new DateTimeType(ZonedDateTime.now()));
+        updateState(SolaxBindingConstants.RAW_DATA, new StringType(data.getRawData()));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Nothing to do here as of now
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        ScheduledFuture<?> schedule = this.schedule;
+        if (schedule != null) {
+            boolean success = schedule.cancel(true);
+            String cancelingSuccessful = success ? "successful" : "failed";
+            logger.debug("Canceling schedule of {} is {}", schedule, cancelingSuccessful);
```

Comment:
In other binding had sometimes issues with schedules which continue run even after handler's disposal. Not sure why this happened, that's why I added this log. Not that it will explain why it happens, rather will confirm the obvious.... probably you're right to delete this.

Reply:
That is usually caused by `cancel(false)` being called rather than `cancel(true)` - or `cancel` not being called at all in `dispose()`, for example because of the reference not being stored correctly.